raiden/tests/integration/test_echo_node.py

```diff
@@ -42,21 +48,17 @@ def test_event_transfer_received_success(token_addresses, raiden_chain, skip_if_
     events = receiver_app.raiden.wal.storage.get_events_by_block(0, 'latest')
     events = [e[1] for e in events]
 
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 1, 'initiator': app0.raiden.address},
-    )
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 2, 'initiator': app1.raiden.address},
-    )
-    assert must_contain_entry(
-        events,
-        EventTransferReceivedSuccess,
-        {'amount': 3, 'initiator': app2.raiden.address},
-    )
+    amounts = [1, 2, 3]
+    addrs = [app0.raiden.address, app1.raiden.address, app2.raiden.address]
+    for amount, addr in zip(amounts, addrs):
+        wait_until(
+            lambda: must_contain_entry(
```

Comment:
This `wait_until` wont work because `events` is static, and not re-evaluated in each loop.
Please, [look here](https://github.com/raiden-network/raiden/pull/1688/files#diff-ba630d95213922afcd778c420e91cfcbR39) and [here](https://github.com/raiden-network/raiden/pull/1636#discussion_r198964958). You should add a context function to re-evaluate the parameters and return the result of `must_contain_entry`, then `assert` the result of `wait_until` over it

Reply:
ah, darn. Thanks, I've updated it.